devtools/maven/src/main/java/io/quarkus/maven/CreateExtensionMojo.java

```diff
@@ -411,123 +558,318 @@ public void execute() throws MojoExecutionException, MojoFailureException {
         }
 
         if (runtimeBomPath != null) {
-            runtimeBomPath = basedir.resolve(runtimeBomPath);
+            runtimeBomPath = basedir.toPath().resolve(runtimeBomPath);
             if (!Files.exists(runtimeBomPath)) {
                 throw new MojoFailureException("runtimeBomPath does not exist: " + runtimeBomPath);
             }
         }
         if (deploymentBomPath != null) {
-            deploymentBomPath = basedir.resolve(deploymentBomPath);
+            deploymentBomPath = basedir.toPath().resolve(deploymentBomPath);
             if (!Files.exists(deploymentBomPath)) {
                 throw new MojoFailureException("deploymentBomPath does not exist: " + deploymentBomPath);
             }
         }
 
-        final Charset charset = Charset.forName(encoding);
-
-        final Path basePomXml = basedir.resolve("pom.xml");
-        if (Files.exists(basePomXml)) {
-            try (Reader r = Files.newBufferedReader(basePomXml, charset)) {
-                Model basePom = new MavenXpp3Reader().read(r);
-                if (!"pom".equals(basePom.getPackaging())) {
+        charset = Charset.forName(encoding);
+
+        try {
+            File rootPom = null;
+            Model rootModel = null;
+            boolean importDeploymentBom = true;
+            boolean setCompilerPluginVersion = true;
+            boolean setQuarkusVersionProp = true;
+            if (isCurrentProjectExists()) {
+                rootPom = getCurrentProjectPom();
+                rootModel = MojoUtils.readPom(rootPom);
+                if (!"pom".equals(rootModel.getPackaging())) {
                     throw new MojoFailureException(
                             "Can add extension modules only under a project with packaging 'pom'; found: "
-                                    + basePom.getPackaging() + "");
+                                    + rootModel.getPackaging() + "");
+                }
+
+                if (rootPom.equals(project.getFile())) {
+                    importDeploymentBom = !hasQuarkusDeploymentBom();
+                    setCompilerPluginVersion = !project.getPluginManagement().getPluginsAsMap()
+                            .containsKey(COMPILER_PLUGIN_KEY);
+                    setQuarkusVersionProp = !project.getProperties().containsKey(QUARKUS_VERSION_PROP);
+                } else {
+                    // aloubyansky: not sure we should support this case and not sure it ever worked properly
```

Comment:
I am also not sure about this... Can we be sure it will work? If not, I think we should focus on the simple use case for now that we know will work.